lib/spotify.rb

```diff
@@ -85,7 +87,17 @@ def tracks_on_album(id)
     end
 
     def who_added_track
-      web.who_added_track(default_user, default_uri, player.current_track_id)
+      web.who_added_track(playlist, player.current_track_id)
+    end
+
+    def playlist
+      # rebuild playist if older than an hour
+      if Time.now - @last_playlist_build > 3600
```

Comment:
This is a bit more complex than I'd like right now. I'd prefer to just cache the entire playlist once (e.g. you'd have to restart the server to reset the cache)

Note that I have a refactor branch here where I'm trying to split out the responsibilities of iterating the playlist and matching the the track id. I'm not sure that's the best approach yet, but I'd like to stick with it for now: https://github.com/jbodah/spotifuby/pull/30

I also really want to work on getting more tests around everything